File: snippets/java/recommend_point_groups.java
import static io.qdrant.client.PointIdFactory.id;

import java.util.List;

import io.qdrant.client.QdrantClient;
import io.qdrant.client.QdrantGrpcClient;

import io.qdrant.client.grpc.Points.RecommendPointGroups;
import io.qdrant.client.grpc.Points.RecommendStrategy;

QdrantClient client =
    new QdrantClient(QdrantGrpcClient.newBuilder("localhost", 6334, false).build());

client.recommendGroupsAsync(RecommendPointGroups.newBuilder()
                .setCollectionName("{collection_name}")
                .setGroupBy("document_id")
                .setGroupSize(2)
                .addAllPositive(List.of(id(100), id(200)))
                .addAllNegative(List.of(id(718)))
                .setStrategy(RecommendStrategy.AverageVector)
                .setLimit(3)
                .build());

File: snippets/rust/recommend_point_groups.rs
use qdrant_client::qdrant::{RecommendPointGroupsBuilder, RecommendStrategy};
use qdrant_client::Qdrant;

let client = Qdrant::from_url("http://localhost:6334").build()?;

client
    .recommend_groups(
        RecommendPointGroupsBuilder::new("{collection_name}", "document_id", 2, 3)
            .positive(100)
            .positive(200)
            .negative(718)
            .strategy(RecommendStrategy::AverageVector),
    )
    .await?;
